src/collectors/hadoop/hadoop.py

```diff
@@ -44,7 +44,12 @@ def get_default_config(self):
         return config
 
     def collect(self):
-        for pattern in self.config['metrics']:
+        if isinstance(self.config['metrics'], list):
```

Comment:
I think a better approach would be to change the code to always take a list of patterns. Something like this:
```
metrics = self.config['metrics']
if not isinstance(metrics, list):
    metrics = [str(metrics)]

for metric in metrics:
    for filename in glob.glob(metric):
        self.collect_from(filename)
```